tests/common.go

```diff
@@ -1002,6 +1004,13 @@ func ValidatePDB(pdbValue int, allowedDisruptions int, initialNumNodes int, isCl
 		} else {
 			currentNumNodes := len(nodes.([]*opsapi.StorageNode))
 			if allowedDisruptions < initialNumNodes-currentNumNodes {
+				log.Warnf("Number of nodes down is more than allowed disruptions [ %d ], total storage nodes : [ %d ] and current count of storage nodes are: [ %d ]", allowedDisruptions, initialNumNodes, currentNumNodes)
+				var currNodesName []string
+				for _, node := range nodes.([]*opsapi.StorageNode) {
+					currNodesName = append(currNodesName, node.SchedulerNodeName)
```

Comment:
Would it return storage nodes which are down or where a node is deleted but node-id is showing down ? Can we also add kubernetes worker node list here.

The issue which we have seen recently was a sequential upgrade and values are not matching..

Reply:
This will return the list of kubernetes nodes where portworx is currently up. Should I return nodes that are down instead? In that case it will be the kubernetes nodes where initially portworx was up and running but currently down.

Reply:
We need all storage nodes where PX is up and down. Also need list of worker nodes currently present in a cluster so that will know where PX is up or down or node delete from the cluster because of the upgrade..

Reply:
I am still not clear, when you say storage node where PX is up/down you mean the node ID? or the kubernetes node name where that px node is run? It is easier to debug and understand using kubernetes node names.
So let me know if this sounds good -

1.  kubernetes node names where PX is running (can also include the node ID)
2. kubernetes node where PX is down (can include node ID)
3. list of all kubernetes nodes

Reply:
Sorry about the confusion. Yes it sounds good. Need both px node id and K8S node name. More like output similar to `pxctl status` command.  

When PDB is not matching what I need to debug is PDB value set in STC and storage nodes(nodes id and name)  in a cluster. Later need to identify if some nodes are recreated or deleted during Anthos upgrade (It can be find by listing k8s nodes)